test/FunctionalTests/E2ETestBase.cs

```diff
@@ -141,5 +142,19 @@ protected virtual void AssertResponse(HttpResponseMessage response, string respo
         {
             throw new NotImplementedException("Must be overriden in derived types.");
         }
+
+        private static int GetNextPort()
```

Comment:
Similar code actually already existed in the IntegrationTests package from Hosting. I've converted the Hosting version to be a copy of the Kestrel one and some of this code will be able to be removed once aspnet/Hosting#996 lands :).